addons/mail/models/mail_thread.py

```diff
@@ -243,6 +237,16 @@ def _compute_message_has_error(self):
     def _search_message_has_error(self, operator, operand):
         return [('message_ids.has_error', operator, operand)]
 
+    @api.multi
+    def _compute_message_attachment_count(self):
+        read_group_var = self.env['ir.attachment'].read_group([('res_model', '=', self._name)],
+                                                              fields=['res_id'],
+                                                              groupby=['res_id'])
+        if len(read_group_var):
```

Comment:
Actually you don't really need the if. Dict comprehension will work even with a void read_group_var (list with 0 elements), and you will always have a value for message_attachment_count with the .get() . So according to me you can remove the if and de-indent the next lines :) .

Reply:
ok thanks, doing the changes